frontend/webapp/containers/main/destinations/add-destination/index.tsx

```diff
@@ -26,31 +24,57 @@ const NotificationNoteWrapper = styled.div`
   margin-top: 24px;
 `;
 
-export function ChooseDestinationContainer() {
-  const [isModalOpen, setModalOpen] = useState(false);
+const AddDestinationButtonWrapper = styled.div`
+  width: 100%;
+  margin-top: 24px;
+`;
+
+const StyledAddDestinationButton = styled(Button)`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  gap: 8px;
+  width: 100%;
+`;
 
+export function ChooseDestinationContainer() {
   const router = useRouter();
-  const { configuredSources, configuredDestinations, resetState } = useAppStore((state) => state);
+  const { createSources, loading: sourcesLoading } = useSourceCRUD();
+  const { createDestination, loading: destinationsLoading } = useDestinationCRUD();
+  const { configuredSources, configuredFutureApps, configuredDestinations, resetState } = useAppStore((state) => state);
 
   const isSourcesListEmpty = () => {
-    const sourceLen = Object.keys(configuredSources).length === 0;
-    if (sourceLen) {
-      return true;
-    }
+    let isEmpty = true;
 
-    let empty = true;
-    for (const source in configuredSources) {
-      if (configuredSources[source].length > 0) {
-        empty = false;
+    const namespaces = Object.keys(configuredSources);
+    if (!namespaces.length) return isEmpty;
+
+    for (const namespace in configuredSources) {
+      if (!!configuredSources[namespace].length) {
+        isEmpty = false;
         break;
       }
     }
-    return empty;
+
+    return isEmpty;
   };
 
+  const [isModalOpen, setModalOpen] = useState(false);
   const handleOpenModal = () => setModalOpen(true);
   const handleCloseModal = () => setModalOpen(false);
 
+  const clickBack = () => {
+    router.push(ROUTES.CHOOSE_SOURCES);
+  };
```

Comment:
wrap this in try catch block, we can show an error message if something went wrong

Reply:
errors are handled by the CRUD hook itself